### PR TITLE
feat: rename to Siera

### PR DESCRIPTION
**Pull request by blu3beri**
- docs: updated README
- docs: string replace agent-cli -> siera
- feat: added version sync
- chore: update LICENSE to Apache-2.0
- chore: added descriptions to every package
- chore: added * version to dependencies
- build: update the git user email to development@animo.id

We likely have to rerelease everything and hope that everything works well.

Next to the name change this also includes some cool cargo workspace functionality that is "bleeding-edge". We can supply all our default metadata in the workspace virtual manifest and inherit it in the other packages. (I have been searching for this feature since I found out about workspaces...)

DO NOT MERGE YET (also need to update the agent-cli-page).

..another note, @morrieinmaas could you check if I did the actions correctly? I did a string replace so I hope its all good.

### CODE REVIEW

File: .github/workflows/crates-release.yml [hunk 1]
```diff
@@ -16,8 +16,11 @@ jobs:
         - uses: actions/checkout@v2
         - uses: actions-rs/toolchain@v1
           with:
-              toolchain: stable
-              override: true
-        - uses: katyo/publish-crates@v1
-          with:
-              registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+            toolchain: stable
+        # We might need to wait some time between publishes
```
Comment (morrieinmaas) [inline .github/workflows/crates-release.yml:20]: Just out of interest: Why would we need to wait?

This looks like the good way to do it as it executes each run as separate script

File: .github/workflows/crates-release.yml [hunk 2]
```diff
@@ -16,8 +16,11 @@ jobs:
         - uses: actions/checkout@v2
         - uses: actions-rs/toolchain@v1
           with:
-              toolchain: stable
-              override: true
-        - uses: katyo/publish-crates@v1
-          with:
-              registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+            toolchain: stable
```
Comment (morrieinmaas) [inline .github/workflows/crates-release.yml:19]: Does this need a login to crates somehow?

File: .github/workflows/crates-release.yml (hunk 1 above)
Comment (blu3beri) [reply .github/workflows/crates-release.yml:20]: SO we do not overload the server, crates.io rate limits. Open for improvements.

File: .github/workflows/crates-release.yml (hunk 2 above)
Comment (blu3beri) [reply .github/workflows/crates-release.yml:19]: WYM?

Comment (morrieinmaas) [reply .github/workflows/crates-release.yml:19]: I menat the code below with `cargo publish`

Comment (blu3beri) [reply .github/workflows/crates-release.yml:19]: Yes, we need an API token to push to crates.io.

Comment (morrieinmaas) [reply .github/workflows/crates-release.yml:19]: How does it pick up the token then now that 
```
     - uses: katyo/publish-crates@v1
       with:
         registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
``` 
is removed?

File: .github/workflows/crates-release.yml (hunk 1 above)
Comment (morrieinmaas) [reply .github/workflows/crates-release.yml:20]: According [this PR about rate limits](https://github.com/rust-lang/crates.io/pull/1596) we should be good.

10 pushes to crates.io anyway and then one per minute

File: .github/workflows/crates-release.yml (hunk 2 above)
Comment (blu3beri) [reply .github/workflows/crates-release.yml:19]: It should read the environment variable `CARGO_REGISTRY_TOKEN`. Unsure if it will work with secrets.

Comment (morrieinmaas) [reply .github/workflows/crates-release.yml:19]: ALright - let's see and find out

**Pull request merged**